### PR TITLE
docs: Clarify the description of `sort-imports` options

### PR DESCRIPTION
**Pull request by gwBear1**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

- [x] Documentation update

#### What changes did you make? (Give an overview)
스크린샷 2024-03-13 오전 10 51 55

---

Made 4 test cases of the below. It could be more straightforward.
```
incorrect with default option ignore false
correct   with default option ignore false
correct   with ignore true
incorrect with ignore true
```
- ignoreCase
- ignoreDeclarationSort
- ignoreMemberSort

#### Is there anything you'd like reviewers to focus on?
- Should be checked whether test cases are correct.

### CODE REVIEW

**Comment by Tanujkanti4441** [comment]
Hi @gwBear1, thanks for the PR, but the docs are correct because it shows both incorrect and correct examples when 
`{ "ignoreCase": true }` as you can see the examples written under the `incorrect` section by opening them in playground.

**Comment by Tanujkanti4441** [comment]
sorry for closing too early.
after looking around a bit i found
```js
/*eslint sort-imports: "error"*/

import A from 'foo.js';
import b from 'bar.js';
```
by default rule doesn't report this arrangement of import names.
```js
/*eslint sort-imports: "error"*/

import a from 'foo.js';
import B from 'bar.js';
```
but report this

@mdjermanovic is it intended

if yes then we need to show this in docs too

**Comment by gwBear1** [comment]
@Tanujkanti4441 Yeah I was confused too. You right, fixing test case would be better choice. I'll rewrite test cases.

**Comment by gwBear1** [comment]
@Tanujkanti4441 And I also think whole document's test case is written wrong way. For example in ignoreDeclarationSort...

스크린샷 2024-03-13 오후 12 48 33

Testing with `{ "ignoreDeclarationSort": true }` is a very useless use case...

I'll update the whole section.

**Comment by gwBear1** [comment]
I added test cases more straightforward. Each options has 4 test cases
- incorrect with default option `ignore false`
- correct with default option `ignore false`
- correct with `ignore true`
- correct2 with `ignore true`

However, I think `ignore true` cases are not useful examples. Those could be deleted if you guys want.

File: docs/src/rules/sort-imports.md [hunk 1]
```diff
@@ -239,18 +252,20 @@ import a from 'bar.js'
...
 
 ::: correct
 
 ```js
-/*eslint sort-imports: ["error", { "ignoreDeclarationSort": true }]*/
-import a from 'foo.js'
-import b from 'bar.js'
+/*eslint sort-imports: ["error", { "ignoreDeclarationSort": false }]*/
+import a from 'bar.js'
+import b from 'foo.js'
```
Comment (Tanujkanti4441) [inline docs/src/rules/sort-imports.md:262]:
```suggestion
import a from 'bar.js';
import b from 'foo.js';
```

there are other examples that don't have semicolon can you please add semicolon to them too?

and also i think its better if we add an incorrect section for `{ "ignoreDeclarationSort": true }` with these examples that shows multiple members import are not allowed by this option:
```js
import {b, a, c} from 'foo.js';
```

File: docs/src/rules/sort-imports.md [hunk 2]
```diff
@@ -278,6 +293,17 @@ import {b, a, c} from 'foo.js'
 
 :::
 
+Examples of **correct** code for this rule with the default `{ "ignoreMemberSort": false }` option:
+
+::: correct
+
+```js
+/*eslint sort-imports: ["error", { "ignoreMemberSort": false }]*/
+import {a, b, c} from 'foo.js'
```
Comment (Tanujkanti4441) [inline docs/src/rules/sort-imports.md:302]:
```suggestion
import {a, b, c} from 'foo.js';
```

and an `incorrect` section can also be added for { "ignoreMemberSort": true } with following example:
```js
import b from 'foo.js';
import a from 'bar.js';
```

File: docs/src/rules/sort-imports.md [hunk 3]
```diff
@@ -193,29 +193,42 @@ import {b, a, c} from 'foo.js'
...
-import a from 'foo.js';
+/*eslint sort-imports: ["error", { "ignoreCase": false }]*/
 import B from 'bar.js';
+import a from 'foo.js';
+import c from 'baz.js';
+```
+
+:::
+
+Examples of **correct** code for this rule with `{ "ignoreCase": true }` option:
```
Comment (Tanujkanti4441) [inline docs/src/rules/sort-imports.md:224]: its fine to keep the `incorrect` example for `{ "ignoreCase": true } ` that shows which case is not allowed by the option.

**Review by Tanujkanti4441 — CHANGES_REQUESTED** [review]

File: docs/src/rules/sort-imports.md (hunk 3 above)
Comment (gwBear1) [reply docs/src/rules/sort-imports.md]: Good suggestion. Thanks for the review!

**Comment by gwBear1** [comment]
@Tanujkanti4441 Can you check my PR again?

File: docs/src/rules/sort-imports.md [hunk 4]
```diff
@@ -193,34 +193,59 @@ import {b, a, c} from 'foo.js'
 ### `ignoreCase`
 
 When `true` the rule ignores the case-sensitivity of the imports local name.
+Uppercase letters of the alphabet must always precede lowercase letters.
```
Comment (Tanujkanti4441) [inline docs/src/rules/sort-imports.md:196]:
```suggestion
When `true` the rule ignores the case-sensitivity of the imports local name.

Uppercase letters of the alphabet must always precede lowercase letters with default settings.
```

better to clear this

**Review by Tanujkanti4441 — CHANGES_REQUESTED** [review]

**Comment by mdjermanovic** [comment]
> ```js
> /*eslint sort-imports: "error"*/
> 
> import A from 'foo.js';
> import b from 'bar.js';
> ```
> 
> 
>     
>       
>     
> 
>       
>     
> 
>     
>   
> by default rule doesn't report this arrangement of import names.
> ```js
> /*eslint sort-imports: "error"*/
> 
> import a from 'foo.js';
> import B from 'bar.js';
> ```
> 
> 
>     
>       
>     
> 
>       
>     
> 
>     
>   
> but report this
> @mdjermanovic is it intended?
> ```

This is intended because: any uppercase letter < any lowercase letter. A < b, B < a.

**Comment by mdjermanovic** [comment]
I'm confused by the PR title. "Fixed incorrect doc description" sounds like the docs were incorrect, but it looks like the docs were correct and we just want to add more examples to further clarify options?

File: docs/src/rules/sort-imports.md [hunk 5]
```diff
@@ -194,33 +194,59 @@ import {b, a, c} from 'foo.js'
 
 When `true` the rule ignores the case-sensitivity of the imports local name.
 
-Examples of **incorrect** code for this rule with the `{ "ignoreCase": true }` option:
+Uppercase letters of the alphabet must always precede lowercase letters with default settings.
```
Comment (mdjermanovic) [inline docs/src/rules/sort-imports.md:197]: If we want to describe both, I think the default behavior should be first.

```md
When `false` (default), uppercase letters of the alphabet must always precede lowercase letters.

When `true`, the rule ignores the case-sensitivity of the imports local name.
```

**Review by gwBear1 — COMMENTED** [review]
fix ignoreCase description

**Comment by gwBear1** [comment]
> I'm confused by the PR title. "Fixed incorrect doc description" sounds like the docs were incorrect, but it looks like the docs were correct and we just want to add more examples to further clarify options?

I changed the PR title and description

**Comment by gwBear1** [comment]
@Tanujkanti4441 @mdjermanovic Followed your comments. Could you check this PR again?

File: docs/src/rules/sort-imports.md [hunk 6]
```diff
@@ -192,35 +192,61 @@ import {b, a, c} from 'foo.js'
...
+
+```js
+/*eslint sort-imports: ["error", { "ignoreCase": true }]*/
+import B from 'foo.js';
+import a from 'bar.js';
+```
+
+:::
+
 Default is `false`.
```
Comment (Tanujkanti4441) [inline docs/src/rules/sort-imports.md:250]: The default values are mentioned in the bottom of the option's section i feel this info should be on top with option's description can you shift them to the top for every option?

**Review by Tanujkanti4441 — COMMENTED** [review]
everything looks good just a last thing.

File: docs/src/rules/sort-imports.md (hunk 6 above)
Comment (gwBear1) [reply docs/src/rules/sort-imports.md]: I'm checking how other rules describe options but all rules have different formats 🤔. Anyway in this document, I'll address your comment.

---
1. `prefer-arrow-callback` - (allowUnboundThis)
They are telling default option as a sentence.
스크린샷 2024-03-21 오후 3 37 04

2. `no-unused-expressions` (allowTernary)
They doesn't talk about default option. It's just defined at the top of the document.
스크린샷 2024-03-21 오후 3 36 50

File: docs/src/rules/sort-imports.md [hunk 7]
```diff
@@ -289,10 +341,22 @@ import {b, a, c} from 'foo.js'
...
+/*eslint sort-imports: ["error", { "ignoreMemberSort": true }]*/
+import b from 'foo.js';
+import a from 'bar.js';
+```
+
+:::
 
 ### `memberSyntaxSortOrder`
 
+Default is `["none", "all", "multiple", "single"]`.
```
Comment (Tanujkanti4441) [inline docs/src/rules/sort-imports.md:358]:
```suggestion
This option takes an array with four predefined elements, the order of elements specifies the order of import styles.

Default order is `["none", "all", "multiple", "single"]`.
```

**Review by Tanujkanti4441 — CHANGES_REQUESTED** [review]

File: docs/src/rules/sort-imports.md [hunk 8]
```diff
@@ -341,10 +405,10 @@ import {a, b} from 'foo.js';
 
 :::
 
-Default is `["none", "all", "multiple", "single"]`.
-
 ### `allowSeparatedGroups`
 
+Default is `false`.
+
 When `true` the rule checks the sorting of import declaration statements only for those that appear on consecutive lines.
```
Comment (Tanujkanti4441) [inline docs/src/rules/sort-imports.md:412]:
```suggestion
When `true`, the rule checks the sorting of import declaration statements only for those that appear on consecutive lines. Default is `false`.
```

**Review by Tanujkanti4441 — CHANGES_REQUESTED** [review]

File: docs/src/rules/sort-imports.md (hunk 6 above)
Comment (Tanujkanti4441) [reply docs/src/rules/sort-imports.md]: Ya! but there is not a particular format for the description, only concept must be clear to the readers.

**Comment by gwBear1** [comment]
@Tanujkanti4441 I hope this would be the last request 😅 Thanks for reviewing be thorough

**Review by Tanujkanti4441 — APPROVED** [review]
LGTM! thanks.

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

**Pull request merged**